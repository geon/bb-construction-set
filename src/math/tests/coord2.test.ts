import { expect, test } from "vitest";
import {
	add,
	Coord2,
	distance,
	interpolate,
	magnitude,
	normalize,
	scale,
	subtract,
} from "../coord2";

test("snapshot", () => {
	const a: Coord2 = { x: 3, y: 4 };
	const b: Coord2 = { x: 30, y: 40 };
	expect([
		add(a, b),
		subtract(a, b),
		scale(a, 3),
		magnitude(a),
		distance(a, b),
		normalize({ x: 2, y: 0 }),
		interpolate(a, b, 0.5),
	]).toMatchSnapshot();
});
