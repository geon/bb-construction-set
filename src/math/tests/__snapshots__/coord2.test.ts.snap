// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot 1`] = `
[
  {
    "x": 33,
    "y": 44,
  },
  {
    "x": -27,
    "y": -36,
  },
  {
    "x": 9,
    "y": 12,
  },
  5,
  45,
  {
    "x": 1,
    "y": 0,
  },
  {
    "x": 16.5,
    "y": 22,
  },
]
`;
